src/middlewares/product.coffee

```diff
@@ -11,7 +11,7 @@ module.exports = (req, res, next) ->
   storeId = req.app.get 'store_id'
 
   res.promises.push(
-    api.get "stores/#{storeId}/products"
+    api.get "stores/#{storeId}/products?order_by=position"
```

Comment:
This path only wants the product matching the specific slug, right? You can do that by

```
GET /X/stores/{store_id}/products?slug={slug}
```

instead, and skip the local filter. More performant + works when the product has a position greater than the max product pagination limit of 100.